Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -60,62 +60,76 @@ void Digitizer::init()
 
 void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
 {
+  digits.clear();
+  mDigits.clear();
+  mMCTruthContainer.clear();
+  
   //array of MCH hits for a given simulated event
+  //to be added: exception handling
   for (auto& hit : hits) {
-    processHit(hit, mEventTime);
+    int labelIndex = mMCTruthContainer.getIndexedSize();//index for this hit
+    int detID = hit.GetDetectorID();
+    if(detID<299){
```

Comment:
something more indicative of the meaning of the comparison would be better, e.g. if ( isStation1(detID) ... )